lib/xandra/protocol.ex

```diff
@@ -18,6 +18,12 @@ defmodule Xandra.Protocol do
     %{frame | body: encode_string_map(requested_options)}
   end
 
+  def encode_request(%Frame{kind: :auth_response} = frame, _requested_options, options) do
+    {authenticator, auth_options} = Keyword.fetch!(options, :authentication)
```

Comment:
You're doing `fetch!` because `:authentication` *must* be in the options since C* asked to authenticate?

If so, @lexmag, should we raise a better error such as "C* wants to auth, but you didn't provide `:authentication`" or something like that?